File: lib/src/common/forms/directives/ng_control.dart
library angular2.src.common.forms.directives.ng_control;

import "control_value_accessor.dart" show ControlValueAccessor;
import "abstract_control_directive.dart" show AbstractControlDirective;
import "package:angular2/src/facade/exceptions.dart" show unimplemented;

/**
 * A base class that all control directive extend.
 * It binds a [Control] object to a DOM element.
 *
 * Used internally by Angular forms.
 */
abstract class NgControl extends AbstractControlDirective {
  String name = null;
  ControlValueAccessor valueAccessor = null;
  Function get validator {
    return unimplemented();
  }

  Function get asyncValidator {
    return unimplemented();
  }

  void viewToModelUpdate(dynamic newValue);
}

File: npm/angular2/es6/prod/src/common/forms/directives/ng_control.js
import { AbstractControlDirective } from './abstract_control_directive';
import { unimplemented } from 'angular2/src/facade/exceptions';
/**
 * A base class that all control directive extend.
 * It binds a {@link Control} object to a DOM element.
 *
 * Used internally by Angular forms.
 */
export class NgControl extends AbstractControlDirective {
    constructor(...args) {
        super(...args);
        this.name = null;
        this.valueAccessor = null;
    }
    get validator() { return unimplemented(); }
    get asyncValidator() { return unimplemented(); }
}
